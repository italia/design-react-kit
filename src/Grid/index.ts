export { GridItem } from './GridItem';
export { GridItemText } from './GridItemText';
export { GridItemTextWrapper } from './GridItemTextWrapper';
export { GridList } from './GridList';
export { GridRow } from './GridRow';

export type { GridItemProps } from './GridItem';
export type { GridItemTextProps } from './GridItemText';
export type { GridItemTextWrapperProps } from './GridItemTextWrapper';
export type { GridListProps } from './GridList';
export type { GridRowProps } from './GridRow';
