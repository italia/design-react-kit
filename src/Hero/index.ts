export { Hero } from './Hero';
export { HeroBackground } from './HeroBackground';
export { HeroBody } from './HeroBody';
export { HeroButton } from './HeroButton';
export { HeroCategory } from './HeroCategory';
export { HeroTitle } from './HeroTitle';

export type { HeroProps } from './Hero';
export type { HeroBackgroundProps } from './HeroBackground';
export type { HeroBodyProps } from './HeroBody';
export type { HeroButtonProps } from './HeroButton';
export type { HeroCategoryProps } from './HeroCategory';
export type { HeroTitleProps } from './HeroTitle';
