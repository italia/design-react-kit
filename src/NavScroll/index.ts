/*
 * This work derives from the React Use Navscroll library
 * Released under the MIT license by Marco Liberati
 * Code: https://github.com/dej611/react-use-navscroll
 */
/* @internal */
export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './types';

export { useNavScroll } from './useNavScroll';
