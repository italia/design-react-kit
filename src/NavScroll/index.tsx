export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './types';

export { useNavScroll } from './useNavScroll';
