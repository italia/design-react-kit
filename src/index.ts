export {
  BreadcrumbItem,
  ButtonGroup,
  ButtonToolbar,
  CardColumns,
  CardDeck,
  CardFooter,
  CardGroup,
  CardHeader,
  CardImg,
  CardImgOverlay,
  CardLink,
  CardSubtitle,
  Col,
  Container,
  Fade,
  Form,
  FormFeedback,
  FormGroup,
  FormText,
  InputGroup,
  InputGroupText,
  Label,
  ListGroup,
  ListGroupItem,
  ListGroupItemHeading,
  ListGroupItemText,
  Media,
  NavItem,
  NavLink,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  PaginationItem as PagerItem,
  PaginationLink as PagerLink,
  Popover,
  PopoverBody,
  PopoverHeader,
  Row,
  TabContent,
  TabPane,
  Table,
  Tooltip,
  UncontrolledAlert,
  UncontrolledCollapse,
  UncontrolledTooltip,
  Util
} from 'reactstrap';

export { Accordion } from './Accordion/Accordion';
export { AccordionBody } from './Accordion/AccordionBody';
export { AccordionHeader } from './Accordion/AccordionHeader';
export { AccordionItem } from './Accordion/AccordionItem';
export { Alert } from './Alert/Alert';
export { AvatarContainer } from './Avatar/AvatarContainer';
export { AvatarExtraText } from './Avatar/AvatarExtraText';
export { AvatarGroupContainer } from './Avatar/AvatarGroupContainer';
export { AvatarIcon } from './Avatar/AvatarIcon';
export { AvatarPresence } from './Avatar/AvatarPresence';
export { AvatarStatus } from './Avatar/AvatarStatus';
export { AvatarWrapper } from './Avatar/AvatarWrapper';
export { BackToTop } from './BackToTop/BackToTop';
export { Badge } from './Badge/Badge';
export { BottomNav } from './BottomNav/BottomNav';
export { BottomNavItem } from './BottomNav/BottomNavItem';
export { Breadcrumb } from './Breadcrumb/Breadcrumb';
export { Button } from './Button/Button';
export { Callout } from './Callout/Callout';
export { CalloutMoreFooter } from './Callout/CalloutMoreFooter';
export { CalloutText } from './Callout/CalloutText';
export { CalloutTitle } from './Callout/CalloutTitle';
export { Card } from './Card/Card';
export { CardBody } from './Card/CardBody';
export { CardCategory } from './Card/CardCategory';
export { CardFooterCTA } from './Card/CardFooterCTA';
export { CardReadMore } from './Card/CardReadMore';
export { CardSignature } from './Card/CardSignature';
export { CardTag } from './Card/CardTag';
export { CardTagsHeader } from './Card/CardTagsHeader';
export { CardText } from './Card/CardText';
export { CardTitle } from './Card/CardTitle';
export { Chip } from './Chips/Chip';
export { ChipLabel } from './Chips/ChipLabel';
export { Collapse } from './Collapse/Collapse';
export { Dimmer } from './Dimmer/Dimmer';
export { DimmerButtons } from './Dimmer/DimmerButtons';
export { Dropdown } from './Dropdown/Dropdown';
export { DropdownMenu } from './Dropdown/DropdownMenu';
export { DropdownToggle } from './Dropdown/DropdownToggle';
export { FontLoader, useFontLoader } from './FontLoader/FontLoader';
export { Forward } from './Forward/Forward';
export { GoBack } from './GoBack/GoBack';
export { GridItem, GridItemText, GridItemTextWrapper, GridList, GridRow } from './Grid/index';
export { Header } from './Header/Header';
export { HeaderBrand } from './Header/HeaderBrand';
export { HeaderContent } from './Header/HeaderContent';
export { HeaderLinkZone } from './Header/HeaderLinkZone';
export { HeaderRightZone } from './Header/HeaderRightZone';
export { HeaderSearch } from './Header/HeaderSearch';
export { HeaderSocialsZone } from './Header/HeaderSocialsZone';
export { HeaderToggler } from './Header/HeaderToggler';
export { Headers } from './Header/Headers';
export { Hero, HeroBackground, HeroBody, HeroButton, HeroCategory, HeroTitle } from './Hero/index';
export { Icon, clearIconCache, iconsList as icons, preloadIcons } from './Icon/Icon';
export { Input } from './Input/Input';
export { InputContainer } from './Input/InputContainer';
export { TextArea } from './Input/TextArea';
export { LinkList } from './LinkList/LinkList';
export { LinkListItem } from './LinkList/LinkListItem';
export { MegamenuFooter } from './Megamenu/MegamenuFooter';
export { MegamenuHighlightColumn } from './Megamenu/MegamenuHighlightColumn';
export { MegamenuItem } from './Megamenu/MegamenuItem';
export { Modal } from './Modal/Modal';
export { ModalBody } from './Modal/ModalBody';
export { ModalFooter } from './Modal/ModalFooter';
export { ModalHeader } from './Modal/ModalHeader';
export { Nav } from './Nav/Nav';
export { useNavScroll } from './NavScroll/index';
export { NotificationManager, createNotification, notify } from './Notification/index';
export { Pager } from './Pager/Pager';
export { Progress } from './Progress/Progress';
export { Rating } from './Rating/Rating';
export { ResponsiveImage } from './ResponsiveImage/ResponsiveImage';
export { Section } from './Section/Section';
export { Select } from './Select/Select';
export { Sidebar } from './Sidebar/Sidebar';
export { Skiplink } from './Skiplink/Skiplink';
export { SkiplinkItem } from './Skiplink/SkiplinkItem';
export { Spinner } from './Spinner/Spinner';
export { StepperContainer } from './Stepper/StepperContainer';
export { StepperContent } from './Stepper/StepperContent';
export { StepperDots } from './Stepper/StepperDots';
export { StepperHeader } from './Stepper/StepperHeader';
export { StepperHeaderElement } from './Stepper/StepperHeaderElement';
export { StepperNav } from './Stepper/StepperNav';
export { ThumbNav } from './ThumbNav/ThumbNav';
export { ThumbNavItem } from './ThumbNav/ThumbNavItem';
export { TimelinePin } from './Timeline/TimelinePin';
export { Timeline } from './Timeline/TimelineWrapper';
export { Toggle } from './Toggle/Toggle';
export { Toolbar } from './Toolbar/Toolbar';
export { ToolbarDividerItem } from './Toolbar/ToolbarDividerItem';
export { ToolbarItem } from './Toolbar/ToolbarItem';

// Types
export type { AccordionProps } from './Accordion/Accordion';
export type { AccordionBodyProps } from './Accordion/AccordionBody';
export type { AccordionHeaderProps } from './Accordion/AccordionHeader';
export type { AlertProps } from './Alert/Alert';
export type { AvatarContainerProps } from './Avatar/AvatarContainer';
export type { AvatarExtraTextProps } from './Avatar/AvatarExtraText';
export type { AvatarGroupContainerProps } from './Avatar/AvatarGroupContainer';
export type { AvatarIconProps } from './Avatar/AvatarIcon';
export type { AvatarPresenceProps } from './Avatar/AvatarPresence';
export type { AvatarStatusProps } from './Avatar/AvatarStatus';
export type { AvatarWrapperProps } from './Avatar/AvatarWrapper';
export type { BadgeProps } from './Badge/Badge';
export type { BottomNavProps } from './BottomNav/BottomNav';
export type { BottomNavItemProps } from './BottomNav/BottomNavItem';
export type { BreadcrumbProps } from './Breadcrumb/Breadcrumb';
export type { ButtonProps } from './Button/Button';
export type { CalloutProps } from './Callout/Callout';
export type { CalloutTextProps } from './Callout/CalloutText';
export type { CalloutTitleProps } from './Callout/CalloutTitle';
export type { CardProps } from './Card/Card';
export type { CardBodyProps } from './Card/CardBody';
export type { CardCategoryProps } from './Card/CardCategory';
export type { CardFooterCTAProps } from './Card/CardFooterCTA';
export type { CardReadMoreProps } from './Card/CardReadMore';
export type { CardSignatureProps } from './Card/CardSignature';
export type { CardTagProps } from './Card/CardTag';
export type { CardTagsHeaderProps } from './Card/CardTagsHeader';
export type { CardTextProps } from './Card/CardText';
export type { CardTitleProps } from './Card/CardTitle';
export type { ChipProps } from './Chips/Chip';
export type { ChipLabelProps } from './Chips/ChipLabel';
export type { DimmerProps } from './Dimmer/Dimmer';
export type { DimmerButtonsProps } from './Dimmer/DimmerButtons';
export type { DropdownProps } from './Dropdown/Dropdown';
export type { DropdownMenuProps } from './Dropdown/DropdownMenu';
export type { DropdownToggleProps } from './Dropdown/DropdownToggle';
export type { FontLoaderProps } from './FontLoader/FontLoader';
export type { ForwardProps } from './Forward/Forward';
export type {
  GridItemProps,
  GridItemTextProps,
  GridItemTextWrapperProps,
  GridListProps,
  GridRowProps
} from './Grid/index';
export type { HeaderProps } from './Header/Header';
export type { HeaderBrandProps } from './Header/HeaderBrand';
export type { HeaderContentProps } from './Header/HeaderContent';
export type { HeaderLinkZoneProps } from './Header/HeaderLinkZone';
export type { HeaderRightZoneProps } from './Header/HeaderRightZone';
export type { HeaderSearchProps } from './Header/HeaderSearch';
export type { HeaderSocialsZoneProps } from './Header/HeaderSocialsZone';
export type { HeaderTogglerProps } from './Header/HeaderToggler';
export type { HeadersProps } from './Header/Headers';
export type {
  HeroBackgroundProps,
  HeroBodyProps,
  HeroButtonProps,
  HeroCategoryProps,
  HeroProps,
  HeroTitleProps
} from './Hero/index';
export type { IconName, IconProps } from './Icon/Icon';
export type { InputProps } from './Input/Input';
export type { InputContainerProps } from './Input/InputContainer';
export type { TextAreaProps } from './Input/TextArea';
export type { LinkListProps } from './LinkList/LinkList';
export type { LinkListItemProps } from './LinkList/LinkListItem';
export type { MegamenuFooterProps } from './Megamenu/MegamenuFooter';
export type { MegamenuHCProps } from './Megamenu/MegamenuHighlightColumn';
export type { MegamenuItemProps } from './Megamenu/MegamenuItem';
export type { ModalProps } from './Modal/Modal';
export type { ModalBodyProps } from './Modal/ModalBody';
export type { ModalFooterProps } from './Modal/ModalFooter';
export type { ModalHeaderProps } from './Modal/ModalHeader';
export type { NavProps } from './Nav/Nav';
export type { ChangesType, RegisterOptions, useNavScrollArgs, useNavScrollResult } from './NavScroll';
export type { NotificationId, NotificationManagerProps, NotificationOptions, NotificationProps } from './Notification';
export type { PagerProps } from './Pager/Pager';
export type { ProgressProps } from './Progress/Progress';
export type { RatingProps } from './Rating/Rating';
export type { ResponsiveImageProps } from './ResponsiveImage/ResponsiveImage';
export type { SectionProps } from './Section/Section';
export type { SelectProps } from './Select/Select';
export type { SidebarProps } from './Sidebar/Sidebar';
export type { SkiplinkProps } from './Skiplink/Skiplink';
export type { SkiplinkItemProps } from './Skiplink/SkiplinkItem';
export type { SpinnerProps } from './Spinner/Spinner';
export type { StepperContainerProps } from './Stepper/StepperContainer';
export type { StepperContentProps } from './Stepper/StepperContent';
export type { StepperDotsProps } from './Stepper/StepperDots';
export type { StepperHeaderProps } from './Stepper/StepperHeader';
export type { StepperHeaderElementProps } from './Stepper/StepperHeaderElement';
export type { StepperNavProps } from './Stepper/StepperNav';
export type { ThumbNavProps } from './ThumbNav/ThumbNav';
export type { ThumbNavItemProps } from './ThumbNav/ThumbNavItem';
export type { TimelinePinProps } from './Timeline/TimelinePin';
export type { TimelineProps } from './Timeline/TimelineWrapper';
export type { ToggleProps } from './Toggle/Toggle';
export type { ToolbarProps } from './Toolbar/Toolbar';
export type { ToolbarItemBadge, ToolbarItemProps } from './Toolbar/ToolbarItem';

export type {
  BreadcrumbItemProps,
  ButtonDropdownProps,
  ButtonGroupProps,
  ButtonToolbarProps,
  CardColumnsProps,
  CardDeckProps,
  CardFooterProps,
  CardGroupProps,
  CardHeaderProps,
  CardImgOverlayProps,
  CardImgProps,
  CardLinkProps,
  CardSubtitleProps,
  ColProps,
  ContainerProps,
  FadeProps,
  FormFeedbackProps,
  FormGroupProps,
  FormProps,
  FormTextProps,
  InputGroupProps,
  InputGroupTextProps,
  LabelProps,
  ListGroupItemHeadingProps,
  ListGroupItemProps,
  ListGroupItemTextProps,
  ListGroupProps,
  MediaProps,
  NavItemProps,
  NavLinkProps,
  NavbarBrandProps,
  NavbarProps,
  NavbarTogglerProps,
  PaginationItem as PagerItemProps,
  PaginationLink as PagerLinkProps,
  PopoverBodyProps,
  PopoverHeaderProps,
  PopoverProps,
  RowProps,
  TabContentProps,
  TabPaneProps,
  TableProps,
  TooltipProps,
  UncontrolledAlertProps,
  UncontrolledCollapseProps,
  UncontrolledTooltipProps
} from 'reactstrap';
