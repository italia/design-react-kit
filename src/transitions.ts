// Taken from Reactstrap utils.js which are tken from bootstrap sass
export const TransitionTimeouts = {
  // $transition-fade
  Fade: 150,
  // $transition-collapse
  Collapse: 350,
  // $modal-transition
  Modal: 300,
  // $carousel-transition
  Carousel: 600
};

// taken from Reactstrap utils.js
export const TransitionsKeys = [
  'in',
  'mountOnEnter',
  'unmountOnExit',
  'appear',
  'enter',
  'exit',
  'timeout',
  'onEnter',
  'onEntering',
  'onEntered',
  'onExit',
  'onExiting',
  'onExited'
];

const transitionStatus = {
  entering: 1,
  entered: 1,
  exiting: 1,
  exited: 1
};

export type TransitionStates = keyof typeof transitionStatus;
