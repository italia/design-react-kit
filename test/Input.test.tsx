import "@testing-library/jest-dom";
import { act, fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import { Input, preloadIcons } from '../src';

// Icons are now async, so preload them to make it behave in an sync way
beforeAll(() => preloadIcons(['it-password-visible', 'it-password-invisible']));

// Test for breaking changes
test('Should support old wrapperClass prop even as @deprecated', () => {
  const { container } = render(<Input wrapperClass='myClass' placeholder='write text here' />);
  expect(container.firstChild).toHaveClass('myClass');
});

test('should display the label', () => {
  const { container } = render(<Input label='MyLabel' />);
  const label = container.querySelector('label');
  if (label) {
    expect(label).toHaveTextContent('MyLabel');
  }
});

test('should display the validationText', () => {
  const { container } = render(<Input validationText='SomeInfo' />);
  const validationText = container.querySelector('small');
  if (validationText) {
    expect(validationText).toHaveTextContent('SomeInfo');
  }
});

test('should display the placeholder', () => {
  const { container } = render(<Input placeholder='Esempio di area di testo' />);
  const input = container.querySelector('input');
  if (input) {
    expect(input).toHaveAttribute('placeholder');
  }
});

test('should make the input readOnly when requested', () => {
  const { container } = render(<Input normalized />);
  const input = container.querySelector('input');
  if (input) {
    expect(input).toHaveAttribute('readOnly');
  }
});

test('should not be normalized when it has a placeholder', () => {
  const { container } = render(<Input normalized placeholder='Esempio di area di testo' />);
  const input = container.querySelector('input');
  if (input) {
    expect(input).not.toHaveAttribute('readOnly');
  }
});

test('should toggle password icon on click', () => {
  const { container } = render(
    <Input
      type='password'
      label='Password'
      id='exampleInputPassword'
      placeholder='Inserisci la tua password'
      wrapperClassName='col col-md-6'
    />
  );
  const icon = container.querySelector('[aria-hidden="true"] > svg');
  const input = container.querySelector('input');
  if (icon && input) {
    act(() => {
      fireEvent.click(icon);
    });
    expect(input).toHaveAttribute('type', 'text');
    act(() => {
      fireEvent.click(icon);
    });
    expect(input).toHaveAttribute('type', 'password');
  }
});

test('should return an unstyled input', () => {
  const { container } = render(<Input noWrapper />);
  expect(container.firstChild?.nodeName).toBe('INPUT');
});

test('should have a testId for resilient UI changes', () => {
  render(<Input testId='test-id-input' />);

  expect(screen.getByTestId('test-id-input')).toBeTruthy();
});
