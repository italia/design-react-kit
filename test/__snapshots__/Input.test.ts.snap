// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories Snapshots Documentazione/Form/Input _InputAutocompleteConDatiAccessibile 1`] = `
<div>
  <div
    class="select-wrapper mb-3"
  >
    <label
      for="autocomplete"
    >
      Provincia
    </label>
    <div
      class="autocomplete__wrapper"
    >
      <div
        style="border: 0px; height: 1px; margin-bottom: -1px; margin-right: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          id="autocomplete__status--A"
          role="status"
        />
        <div
          aria-atomic="true"
          aria-live="polite"
          id="autocomplete__status--B"
          role="status"
        />
      </div>
      
      <input
        aria-activedescendant="false"
        aria-autocomplete="list"
        aria-describedby="autocomplete__assistiveHint"
        aria-expanded="false"
        aria-owns="autocomplete__listbox"
        autocomplete="off"
        class="autocomplete__input autocomplete__input--default"
        id="autocomplete"
        name="input-autocomplete"
        placeholder="Testo da cercare"
        role="combobox"
        type="text"
        value=""
      />
      <ul
        class="autocomplete__menu autocomplete__menu--inline autocomplete__menu--hidden"
        id="autocomplete__listbox"
        role="listbox"
      />
      <span
        id="autocomplete__assistiveHint"
        style="display: none;"
      >
        When autocomplete results are available use up and down arrows to review and enter to select.  Touch device users, explore by touch or with swipe gestures.
      </span>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input AreaDiTesto 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=" "
    >
      Esempio di area di testo
    </label>
    <textarea
      class=""
      rows="3"
    />
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input AreaDiTestoConSegnaposto 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class="active "
      for="example-textarea-placeholder"
    >
      Esempio di area di testo
    </label>
    <textarea
      aria-describedby="example-textarea-placeholderDescription"
      class=""
      id="example-textarea-placeholder"
      placeholder="Testo di esempio"
      rows="3"
    />
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input Disabilitato 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=" "
      for="exampleDisabled"
    >
      Contenuto disabilitato
    </label>
    <input
      aria-describedby="exampleDisabledDescription"
      class="form-control"
      disabled=""
      id="exampleDisabled"
      type="text"
      value=""
    />
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input EsempiDiCampiDiInput 1`] = `
<div>
  <div>
    <div
      class="form-group"
    >
      <label
        class=" "
        for="exampleInputText"
      >
        Campo di tipo testuale
      </label>
      <input
        aria-describedby="exampleInputTextDescription"
        class="form-control"
        id="exampleInputText"
        type="text"
        value=""
      />
      <div
        class="form-text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=" "
        for="exampleInputEmail"
      >
        Campo di tipo email
      </label>
      <input
        aria-describedby="exampleInputEmailDescription"
        class="form-control"
        id="exampleInputEmail"
        type="email"
        value=""
      />
      <div
        class="form-text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=" input-number-label"
        for="exampleInputNumber"
      >
        Campo di tipo numerico
      </label>
      <div
        class="input-group input-number"
      >
        <input
          aria-describedby="exampleInputNumberDescription"
          class="form-control"
          id="exampleInputNumber"
          type="number"
          value=""
        />
        <span
          class="input-group-text align-buttons flex-column"
        >
          <button
            class="input-number-add"
            type="button"
          >
            <span
              class="visually-hidden"
            />
          </button>
          <button
            class="input-number-sub"
            type="button"
          >
            <span
              class="visually-hidden"
            />
          </button>
        </span>
      </div>
      <div
        class="form-text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class=" "
        for="exampleInputTel"
      >
        Campo di tipo telefono
      </label>
      <input
        aria-describedby="exampleInputTelDescription"
        class="form-control"
        id="exampleInputTel"
        type="tel"
        value=""
      />
      <div
        class="form-text"
      />
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input InputConIconaOBottoni 1`] = `
<div>
  <div>
    <div
      class="form-group"
    >
      <div
        class="input-group"
      >
        <span
          class="input-group-text"
        >
          <svg
            aria-hidden="true"
            class="icon icon-sm"
            enable-background="new 0 0 24 24"
            role="img"
            viewBox="0 0 24 24"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              icon
            </title>
            <path
              d="M20.5 3.5c-.5-.6-1.3-.9-2.1-.9s-1.6.3-2.2.9L4.6 15.2l-1 5.2 5.2-1L20.5 7.8c1.2-1.2 1.2-3.1 0-4.3zm-5.7 2.8.7-.7 2.9 2.9-.7.7-2.9-2.9zm-.7.8 1.1 1-8.4 8.4-1.1-1.1 8.4-8.3zM5.7 18.3c-.1-.2-.4-.3-.6-.3l.3-1.5 2.1 2.1-1.5.3c0-.2-.1-.5-.3-.6zm2.9 0-1.1-1.1 8.4-8.4 1 1.1-8.3 8.4zM19.8 7.1l-.7.7-2.9-2.9.7-.7c.4-.4.9-.6 1.5-.6.5 0 1 .2 1.4.6.8.8.8 2.1 0 2.9z"
            />
          </svg>
        </span>
        <label
          class=""
          for="input-group-1"
        >
          Con Etichetta
        </label>
        <input
          class="form-control"
          id="input-group-1"
          name="input-group-1"
          type="text"
        />
      </div>
    </div>
    <div
      class="form-group"
    >
      <div
        class="input-group"
      >
        <span
          class="input-group-text"
        >
          <svg
            aria-hidden="true"
            class="icon icon-danger icon-sm"
            enable-background="new 0 0 24 24"
            role="img"
            viewBox="0 0 24 24"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              icon
            </title>
            <path
              d="M20.5 3.5c-.5-.6-1.3-.9-2.1-.9s-1.6.3-2.2.9L4.6 15.2l-1 5.2 5.2-1L20.5 7.8c1.2-1.2 1.2-3.1 0-4.3zm-5.7 2.8.7-.7 2.9 2.9-.7.7-2.9-2.9zm-.7.8 1.1 1-8.4 8.4-1.1-1.1 8.4-8.3zM5.7 18.3c-.1-.2-.4-.3-.6-.3l.3-1.5 2.1 2.1-1.5.3c0-.2-.1-.5-.3-.6zm2.9 0-1.1-1.1 8.4-8.4 1 1.1-8.3 8.4zM19.8 7.1l-.7.7-2.9-2.9.7-.7c.4-.4.9-.6 1.5-.6.5 0 1 .2 1.4.6.8.8.8 2.1 0 2.9z"
            />
          </svg>
        </span>
        <label
          class="active"
          for="input-group-2"
        >
          Con Etichetta e placeholder
        </label>
        <input
          class="form-control"
          id="input-group-2"
          name="input-group-2"
          placeholder="Lorem Ipsum"
          type="text"
        />
      </div>
    </div>
    <div
      class="form-group"
    >
      <div
        class="input-group"
      >
        <span
          class="input-group-text"
        >
          <svg
            aria-hidden="true"
            class="icon icon-primary icon-sm"
            enable-background="new 0 0 24 24"
            role="img"
            viewBox="0 0 24 24"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              icon
            </title>
            <path
              d="M20.5 3.5c-.5-.6-1.3-.9-2.1-.9s-1.6.3-2.2.9L4.6 15.2l-1 5.2 5.2-1L20.5 7.8c1.2-1.2 1.2-3.1 0-4.3zm-5.7 2.8.7-.7 2.9 2.9-.7.7-2.9-2.9zm-.7.8 1.1 1-8.4 8.4-1.1-1.1 8.4-8.3zM5.7 18.3c-.1-.2-.4-.3-.6-.3l.3-1.5 2.1 2.1-1.5.3c0-.2-.1-.5-.3-.6zm2.9 0-1.1-1.1 8.4-8.4 1 1.1-8.3 8.4zM19.8 7.1l-.7.7-2.9-2.9.7-.7c.4-.4.9-.6 1.5-.6.5 0 1 .2 1.4.6.8.8.8 2.1 0 2.9z"
            />
          </svg>
        </span>
        <label
          class=""
          for="input-group-3"
        >
          Con Etichetta e bottone di tipo primary
        </label>
        <input
          class="form-control"
          id="input-group-3"
          name="input-group-3"
          type="text"
        />
        <div
          class="input-group-append"
        >
          <button
            class="btn btn-primary"
            id="button-3"
            type="button"
          >
            Invio
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input InputDatepicker 1`] = `
<div>
  <div
    class="active form-group"
  >
    <label
      class="active "
    >
      Datepicker
    </label>
    <input
      class="active form-control"
      placeholder="22/12/2023"
      type="date"
      value=""
    />
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input InputHourpicker 1`] = `
<div>
  <div
    class="active form-group"
  >
    <label
      class="active "
    >
      Hourpicker
    </label>
    <input
      class="active form-control"
      type="time"
      value=""
    />
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input InputPassword 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=" "
      for="exampleInputPassword"
    >
      Password con label, placeholder e testo di aiuto
    </label>
    <input
      aria-describedby="exampleInputPasswordDescription"
      class="form-control"
      id="exampleInputPassword"
      type="password"
      value=""
    />
    <span
      aria-hidden="true"
      class="password-icon"
    >
      <svg
        class="icon password-icon-visible icon-sm"
        enable-background="new 0 0 24 24"
        role="img"
        viewBox="0 0 24 24"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          icon
        </title>
        <path
          d="M21.8 11.5C19 6.1 12.4 3.9 7 6.7c-2.1 1.1-3.7 2.7-4.8 4.8L2 12l.2.5c2.8 5.4 9.4 7.6 14.8 4.8 2.1-1.1 3.7-2.7 4.8-4.8l.2-.5-.2-.5zm-9.8 6c-3.8 0-7.2-2.1-8.9-5.5C4.8 8.6 8.2 6.5 12 6.5c3.8 0 7.2 2.1 8.9 5.5a9.88 9.88 0 0 1-8.9 5.5zM12 8c-2.2 0-4 1.8-4 4s1.8 4 4 4 4-1.8 4-4-1.8-4-4-4zm0 7c-1.7 0-3-1.3-3-3s1.3-3 3-3 3 1.3 3 3c.1 1.6-1.2 2.9-2.8 3H12z"
        />
      </svg>
    </span>
    <small
      class="form-text"
      id="exampleInputPasswordDescription"
    >
      Inserisci almeno 8 caratteri e una lettera maiuscola
    </small>
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input Readonly 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=" "
      for="examplereadOnly"
    >
      Contenuto in sola lettura
    </label>
    <input
      aria-describedby="examplereadOnlyDescription"
      class="form-control"
      id="examplereadOnly"
      readonly=""
      type="text"
      value=""
    />
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input ReadonlyNormalizzato 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=" "
      for="exampleNormalized"
    >
      Contenuto in sola lettura
    </label>
    <input
      aria-describedby="exampleNormalizedDescription"
      class="form-control-plaintext"
      id="exampleNormalized"
      readonly=""
      type="text"
      value=""
    />
    <div
      class="form-text"
    />
  </div>
</div>
`;

exports[`Stories Snapshots Documentazione/Form/Input UtilizzoDiPlaceholderELabel 1`] = `
<div>
  <div>
    <div
      class="form-group"
    >
      <label
        class=" "
        for="exampleLabel"
      >
        Etichetta di esempio
      </label>
      <input
        aria-describedby="exampleLabelDescription"
        class="form-control"
        id="exampleLabel"
        type="text"
        value=""
      />
      <div
        class="form-text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="active "
        for="examplePlaceholder"
      >
        Etichetta di esempio
      </label>
      <input
        aria-describedby="examplePlaceholderDescription"
        class="form-control"
        id="examplePlaceholder"
        placeholder="Testo di esempio"
        type="text"
        value=""
      />
      <div
        class="form-text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="active "
        for="exampleInfoText"
      >
        Etichetta di esempio
      </label>
      <input
        aria-describedby="exampleInfoTextDescription"
        class="form-control"
        id="exampleInfoText"
        placeholder="Testo di esempio"
        type="text"
        value=""
      />
      <small
        class="form-text"
        id="exampleInfoTextDescription"
      >
        Ulteriore testo informativo
      </small>
      <div
        class="form-text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="active "
        for="examplevalidationText"
      >
        Etichetta di esempio
      </label>
      <input
        aria-describedby="examplevalidationTextDescription"
        class="is-invalid form-control"
        id="examplevalidationText"
        placeholder="Testo di esempio"
        type="text"
        value=""
      />
      <div
        class="invalid-feedback form-feedback just-validate-error-label form-text form-feedback just-validate-error-label"
      >
        Campo non valido
      </div>
    </div>
  </div>
</div>
`;
